processing/src/main/java/io/druid/segment/data/CompressedIntsIndexedSupplier.java

```diff
@@ -365,5 +366,13 @@ public void close() throws IOException
     {
       Closeables.close(holder, false);
     }
+
+    @Override
+    public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+    {
+      // ideally should inspect buffer, but at the moment of inspectRuntimeShape() call buffer might be null although
```

Comment:
Didn't understand this question. However any implementation of `inspectRuntimeShape()` anywhere doesn't affect correctness in any way.

The purpose of `inspectRuntimeShape()` is to ensure that some copy of the code is called always with the same runtime shape (monomorphic), and (ideally) there is only one copy of the code for each runtime shape, not to make JIT perform the same compilation work twice and waste code cache (on Hotspot level) and instruction cache (on hardware level).

Not reflecting all fields which actually belong to runtime shape in `inspectRuntimeShape()` could lead to the situation when some copy of the code is called with non-monomorphic runtime shape, that will make Hotspot JIT to generate slower (but still correct) code.

Reflecting some fields which actually don't belong to runtime shape in `inspectRuntimeShape()` could lead to the situation when there are several copies of the code which are called with the same runtime shape, that will pollute code cache and will make JIT to perform the same work several times, but the code is still correct.